libs/agno/agno/playground/router.py

```diff
@@ -411,54 +473,105 @@ async def process_image(file: UploadFile) -> List[Union[str, Dict]]:
         return [encoded, image_info]
 
     @playground_router.post("/agent/run")
-    async def agent_run(body: AgentRunRequest):
-        logger.debug(f"AgentRunRequest: {body}")
-        agent = get_agent_by_id(body.agent_id, agents)
+    async def agent_run(
+        message: str = Form(...),
+        agent_id: str = Form(...),
+        stream: bool = Form(True),
+        monitor: bool = Form(False),
+        session_id: Optional[str] = Form(None),
+        user_id: Optional[str] = Form(None),
+        files: Optional[List[UploadFile]] = File(None),
+        image: Optional[UploadFile] = File(None),
+    ):
+        logger.debug(f"AgentRunRequest: {message} {session_id} {user_id} {agent_id}")
+        agent = get_agent_by_id(agent_id, agents)
         if agent is None:
             raise HTTPException(status_code=404, detail="Agent not found")
 
-        if body.session_id is not None:
-            logger.debug(f"Continuing session: {body.session_id}")
+        if files:
+            if agent.knowledge is None:
+                raise HTTPException(status_code=404, detail="KnowledgeBase not found")
+
+        if session_id is not None:
+            logger.debug(f"Continuing session: {session_id}")
         else:
             logger.debug("Creating new session")
 
         # Create a new instance of this agent
-        new_agent_instance = agent.deep_copy(update={"session_id": body.session_id})
-        if body.user_id is not None:
-            new_agent_instance.user_id = body.user_id
+        new_agent_instance = agent.deep_copy(update={"session_id": session_id})
+        if user_id is not None:
+            new_agent_instance.user_id = user_id
 
-        if body.monitor:
+        if monitor:
             new_agent_instance.monitoring = True
         else:
             new_agent_instance.monitoring = False
 
         base64_image: Optional[List[Union[str, Dict]]] = None
-        if body.image:
-            base64_image = await process_image(body.image)
-
-        if body.stream:
+        if image:
```

Comment:
Lets move duplicate code like this into helpers. So the routers file doesn't get so long just for duplicate code.